src/loopfilter.rs

```diff
@@ -1000,3 +1008,132 @@ unsafe fn loop_filter_v_sb128uv_rust<BD: BitDepth>(
+
+                if !flags.contains(CpuFlags::AVX512ICL) {
+                    return;
+                }
+
+                (*c).loop_filter_sb[0][0] = dav1d_lpf_h_sb_y_16bpc_avx512icl;
+                (*c).loop_filter_sb[0][1] = dav1d_lpf_v_sb_y_16bpc_avx512icl;
+                (*c).loop_filter_sb[1][0] = dav1d_lpf_h_sb_uv_16bpc_avx512icl;
+                (*c).loop_filter_sb[1][1] = dav1d_lpf_v_sb_uv_16bpc_avx512icl;
+            }
+        }
+    }
+}
+
+#[cfg(all(feature = "asm", any(target_arch = "arm", target_arch = "aarch64")))]
+#[inline(always)]
+unsafe fn loop_filter_dsp_init_arm<BD: BitDepth>(c: *mut Rav1dLoopFilterDSPContext) {
+    use crate::include::common::bitdepth::BPC;
+
+    let flags = rav1d_get_cpu_flags();
+
+    if !flags.contains(CpuFlags::NEON) {
+        return;
+    }
+
+    match BD::BPC {
+        BPC::BPC8 => {
+            (*c).loop_filter_sb[0][0] = dav1d_lpf_h_sb_y_8bpc_neon;
+            (*c).loop_filter_sb[0][1] = dav1d_lpf_v_sb_y_8bpc_neon;
+            (*c).loop_filter_sb[1][0] = dav1d_lpf_h_sb_uv_8bpc_neon;
+            (*c).loop_filter_sb[1][1] = dav1d_lpf_v_sb_uv_8bpc_neon;
+        }
+        BPC::BPC16 => {
+            (*c).loop_filter_sb[0][0] = dav1d_lpf_h_sb_y_16bpc_neon;
+            (*c).loop_filter_sb[0][1] = dav1d_lpf_v_sb_y_16bpc_neon;
+            (*c).loop_filter_sb[1][0] = dav1d_lpf_h_sb_uv_16bpc_neon;
+            (*c).loop_filter_sb[1][1] = dav1d_lpf_v_sb_uv_16bpc_neon;
+        }
+    }
+}
+
+#[cold]
+pub unsafe fn rav1d_loop_filter_dsp_init_8bpc(c: *mut Rav1dLoopFilterDSPContext) {
+    (*c).loop_filter_sb[0][0] = loop_filter_h_sb128y_c_erased::<BitDepth8>;
+    (*c).loop_filter_sb[0][1] = loop_filter_v_sb128y_c_erased::<BitDepth8>;
+    (*c).loop_filter_sb[1][0] = loop_filter_h_sb128uv_c_erased::<BitDepth8>;
+    (*c).loop_filter_sb[1][1] = loop_filter_v_sb128uv_c_erased::<BitDepth8>;
+
+    #[cfg(feature = "asm")]
+    cfg_if! {
+        if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+            loop_filter_dsp_init_x86::<BitDepth8>(c);
+        } else if #[cfg(any(target_arch = "arm", target_arch = "aarch64"))] {
+            loop_filter_dsp_init_arm::<BitDepth8>(c);
+        }
+    }
+}
+
+#[cold]
+pub unsafe fn rav1d_loop_filter_dsp_init_16bpc(c: *mut Rav1dLoopFilterDSPContext) {
+    (*c).loop_filter_sb[0][0] = loop_filter_h_sb128y_c_erased::<BitDepth16>;
+    (*c).loop_filter_sb[0][1] = loop_filter_v_sb128y_c_erased::<BitDepth16>;
+    (*c).loop_filter_sb[1][0] = loop_filter_h_sb128uv_c_erased::<BitDepth16>;
+    (*c).loop_filter_sb[1][1] = loop_filter_v_sb128uv_c_erased::<BitDepth16>;
+
+    #[cfg(feature = "asm")]
+    cfg_if! {
+        if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+            loop_filter_dsp_init_x86::<BitDepth16>(c);
+        } else if #[cfg(any(target_arch = "arm", target_arch = "aarch64"))] {
+            loop_filter_dsp_init_arm::<BitDepth16>(c);
+        }
+    }
+}
```

Comment:
This is a fine intermediate state, but this doesn't actually deduplicate `fn rav1d_loop_filter_dsp_init`.